### PR TITLE
Add more battery operation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- Charge: Add shipmode into battery_ioctl.h
- Charge: Add cut-off current interface 

## Impact

New operation

## Testing

Xiaomi Watch S1 Pro

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Maybe it worth of a single commit?

**Comment by xiaoxiang781216** [comment]
> Maybe it worth of a single commit?

The requirement come one by one:(.

**Review by acassis — APPROVED** [review]

File: include/nuttx/power/battery_ioctl.h [hunk 1]
```diff
@@ -143,6 +143,7 @@ enum batio_operate_e
   BATIO_OPRTN_SYSON,
   BATIO_OPRTN_RESET,
   BATIO_OPRTN_WDOG,
+  BATIO_OPRTN_SHIPMODE,
```
Comment (acassis) [inline include/nuttx/power/battery_ioctl.h:146]: I was going to suggest a generic name like _STANDBY, but I agree _SHIPMODE is has a sense of purpose. In the automotive industry we use it too when shipping a car from a country to another or even inside big countries

File: include/nuttx/power/battery_ioctl.h [hunk 2]
```diff
@@ -144,6 +144,7 @@ enum batio_operate_e
   BATIO_OPRTN_RESET,
   BATIO_OPRTN_WDOG,
   BATIO_OPRTN_SHIPMODE,
+  BATIO_OPRTN_CUTOFF_CURRENT,
```
Comment (acassis) [inline include/nuttx/power/battery_ioctl.h:147]: Isn't BATIO_OPRTN_CUTOFF_CURRENT equivalent to BATIO_OPRTN_SYSOFF or even BATIO_OPRTN_HIZ ? What is the purpose in this case?

File: include/nuttx/power/battery_ioctl.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/power/battery_ioctl.h:146]: The standby is too general and may give the programmer the expression that the charger will switch between standby and normal frequently. But the ship mode is set on factory line, exit when the customer activates the device at first time, and then never come back to the ship mode.

**Pull request merged**